services/tables/src/main/java/com/linkedin/openhouse/tables/config/MainApplicationConfig.java

```diff
@@ -74,6 +81,32 @@ public UserTableApi provideApiInstance() {
     return new UserTableApi(apiClient);
   }
 
+  @Bean
+  public WebMvcTagsContributor clientIdTagContributor() {
```

Comment:
this is the core piece of this PR: takes header from client request and sets it into the metrics registry tags for use in the handler interceptor for metric emission

Reply:
actually, we may want to have a primary bean within LI and there we puts constraints to only allow spark and trino

Reply:
why primary? if i understand correctly, if beans collide then its a a startup exception. isn't that desirable? otherwise user may have a bean that doesn't get registered and have to debug. 

>actually, we may want to have a primary bean within LI and there we puts constraints to only allow spark and trino

ok good idea. i'm thinking to add the 
`public static final List<String> ALLOWED_CLIENT_NAME_VALUES = Arrays.asList("trino", "spark");` field as LI bean, and default to an empty array if not provided. since all OS users may benefit from this header. what do you think?

Reply:
that sounds fine also, to have a separate bean.